pulser/pulse.py

```diff
@@ -136,17 +156,17 @@ def draw(self):
         fig.tight_layout()
         plt.show()
 
-    def _to_dict(self):
+    def _to_dict(self) -> Dict[str, Union[float, Waveform]]:
```

Comment:
I think the `Union` inside the `Dict` args don't really work as expected. Just put `Dict[str, Any]`